src/transactions/test/MergeTests.cpp

```diff
@@ -733,28 +759,45 @@ TEST_CASE("merge", "[tx][merge]")
                     ltx.commit();
                 }
 
-                auto merge = [&](bool addSigner) {
+                auto merge = [&](bool addSigner, AccountID const& dest) {
                     if (addSigner)
                     {
                         addSponsoredSigner(
                             acc1, 0, &sponsoringAcc.getPublicKey(), 0, 2, 3, 0);
                     }
 
-                    acc1.merge(b1);
+                    if (ledgerVersion < 16 &&
+                        dest == sponsoringAcc.getPublicKey())
+                    {
+                        REQUIRE_THROWS(acc1.merge(dest));
 
-                    LedgerTxn ltx(app->getLedgerTxnRoot());
-                    checkSponsorship(ltx, sponsoringAcc.getPublicKey(), 0,
-                                     nullptr, 0, 2, 0, 0);
+                        LedgerTxn ltx(app->getLedgerTxnRoot());
+                        checkSponsorship(ltx, sponsoringAcc.getPublicKey(), 0,
```

Comment:
This is a bit confusing. I stared at this for a while before recognizing why it works (I even got to the point that I started a build to actually run the test). The key observation is that we never call with `addSigner && dest == sponsoringAcc.getPublicKey()`. For clarity, it might be worth either (a) adding a comment to explain that we are explicitly not handling that case or (b) handling that case.

Reply:
Good catch. Added a section to handle the missing case.